### PR TITLE
Building with Clang on Windows

### PR DESCRIPTION
**Pull request by shibatch**
With this patch, the library can be built with Clang on Windows.

### CODE REVIEW

File: Configure.cmake [hunk 1]
```diff
@@ -32,25 +25,34 @@ if (ENFORCE_TESTER3 AND NOT SLEEF_OPENSSL_FOUND)
...
-if (LIBFFTW3)
-  find_path(FFTW3_INCLUDE_DIR
-    NAMES fftw3.h
-    ONLY_CMAKE_FIND_ROOT_PATH)
-endif(LIBFFTW3)
-
-if (NOT LIBM)
-  set(LIBM "")
-endif()
+if (NOT SLEEF_CLANG_ON_WINDOWS)
```
Comment (fpetrogalli) [inline Configure.cmake:28]: I don't get this - why do you need to customize this for clang on windows? If the libraryes are not available, the variables below (`LIB_MPFR`, `LIBFFTW3`, ..) would evaluate to false and teh corresponding compilation flags will not be set. Is that breaking things?

File: Configure.cmake [hunk 2]
```diff
@@ -310,6 +312,10 @@ if(CMAKE_C_COMPILER_ID MATCHES "(GNU|Clang)")
     string(CONCAT FLAGS_WALL ${FLAGS_WALL} " -Wno-psabi")
     set(FLAGS_ENABLE_NEON32 "-mfpu=neon")
   endif(CMAKE_C_COMPILER_ID MATCHES "GNU")
+
+  if (SLEEF_CLANG_ON_WINDOWS)
+    string(CONCAT FLAGS_WALL ${FLAGS_WALL} " -D_CRT_SECURE_NO_WARNINGS -Wno-deprecated-declarations")
```
Comment (fpetrogalli) [inline Configure.cmake:317]: Why is this needed?

Comment (shibatch) [reply Configure.cmake:317]: Because clang references MSVC header files.
They have deprecation and security attributes for many functions, and many deprecation warnings are displayed without these options.

File: Configure.cmake (hunk 1 above)
Comment (shibatch) [reply Configure.cmake:28]: CMake finds libraries for Cygwin, and this causes build errors.
We need to check the builds with Cygwin testers.

Comment (fpetrogalli) [reply Configure.cmake:28]: May I ask you to add this explanation as a comment? It will help people like me who doesn't know much about compiling on windows in understanding why there is this `if not clang_on_windows` condition.

File: Configure.cmake (hunk 2 above)
Comment (fpetrogalli) [reply Configure.cmake:317]: Please explain this with a comment in the sources too.

File: src/quad/CMakeLists.txt [hunk 3]
```diff
@@ -60,7 +60,7 @@ foreach(SIMD ${SLEEFQUAD_SUPPORTED_EXT})
   endif()
 endforeach()
 
-if(MSVC OR MINGW AND WIN32)
+if(MSVC OR SLEEF_CLANG_ON_WINDOWS OR MINGW AND WIN32)
```
Comment (fpetrogalli) [inline src/quad/CMakeLists.txt:63]: Why is this condition different from the one `if((MSVC OR MINGW AND WIN32) OR SLEEF_CLANG_ON_WINDOWS)` 25 of `src/libm/CMakeLists.txt`?

File: Configure.cmake [hunk 4]
```diff
@@ -2,12 +2,9 @@ include(CheckCCompilerFlag)
...
 
-# Some toolchains require explicit linking of the libraries following.
-find_library(LIB_MPFR mpfr)
-find_library(LIBM m)
-find_library(LIBGMP gmp)
-find_library(LIBRT rt)
-find_library(LIBFFTW3 fftw3)
+# Detect if cmake is running on CI environments
+string(COMPARE NOTEQUAL "" "$ENV{TRAVIS}" RUNNING_ON_TRAVIS)
+string(COMPARE NOTEQUAL "" "$ENV{APPVEYOR}" RUNNING_ON_APPVEYOR)
```
Comment (fpetrogalli) [inline Configure.cmake:7]: Please make these variables to be set at `cmake` invocation, as in `cmake -DRUNNING_ON_TRAVIS=On ...`. It is safer than having to deal with things breaking in the case the name of the env variable change.

Comment (shibatch) [reply Configure.cmake]: Done.

File: src/quad/CMakeLists.txt (hunk 3 above)
Comment (shibatch) [reply src/quad/CMakeLists.txt]: Those two are now same.

File: Configure.cmake (hunk 1 above)
Comment (shibatch) [reply Configure.cmake]: I added the comment.
